tests/gtest/cppapi_test.cpp

```diff
@@ -43,15 +43,9 @@ std::map<Samples10Points, std::vector<float>> samples_10_data(float fill)
 
 class SubvolumeTest : public ::testing::Test {
 protected:
-    void SetUp() override {
-        datahandle = make_single_datahandle(SAMPLES_10.c_str(), CREDENTIALS.c_str());
-    }
-
-    void TearDown() override {
-        delete datahandle;
-    }
+    SubvolumeTest() : datahandle(make_single_datahandle(SAMPLES_10.c_str(), CREDENTIALS.c_str())) {}
```

Comment:
This changes the behaviour of this class beyond just switching from a heap allocated SingleDataHandle. Now a new data handle is created at construction and destroyed at destruction of the SubVolumeTest class, rather than using the setup and teardown methods.

I have not looked into the details of how this class is used, so I don't know if this breaks anything. Was this intentional?

Reply:
To all of your comments:
yes, it was intentional because otherwise there were issues with compilation (member initialization).

According to gtest [documentation](https://github.com/google/googletest/blob/main/docs/faq.md#should-i-use-the-constructordestructor-of-the-test-fixture-or-setupteardown-ctorvssetup) that is almost the same, so I didn't bother put it into spotlight.